### ISSUE TITLE
Standardize `max_length` globally across all models

### ISSUE THREAD

**Issue by jathanism** [nautobot/nautobot#2906]
### Proposed Changes

The `max_length` field varies widley. This `grep` catches most of it (with uniqueness counts):

```
$ grep -r " name = .*\models.CharField" nautobot | grep -o "max_length.*\d+" | sort -n | uniq -c
  37 max_length=100
   1 max_length=150
   1 max_length=20
   1 max_length=200
   2 max_length=255
   5 max_length=50
   7 max_length=64
```

### Justification

This inconsistency causes recurring confusion in UX at a minimum, not to mention complicates testing and validation often enough to be a pain.

NB: This may be solved by the introduction of `BasePropertiesModel` or similar in #2822?

# TODO

- [x] #2603 
- [x] #3171

**Comment by glennmatthews** [comment]
Should revisit this aspect of our data models more generally - does `description` need to be limited to 200 characters or should it also increase, etc.

**Comment by bryanculver** [comment]
Should consider `CustomFieldChoice`'s `value` as well.

**Comment by lampwins** [comment]
Will want to doublecheck forms and serializer fields

**Comment by glennmatthews** [comment]
Implemented by #5355.
